app-frontend/src/app/core/services/map.service.js

```diff
@@ -0,0 +1,324 @@
+import Map from 'es6-map';
+/* eslint no-underscore-dangle: ["error", {"allowAfterThis": true}] */
+/* eslint no-unused-vars: 0 */
+/* eslint spaced-comment: 0 */
+
+class MapWrapper {
+    constructor(leafletMap, mapId, options) {
+        this.map = leafletMap;
+        this.mapId = mapId;
+        // counter is not global - each map has its own.
+        // So don't combine lists of listeners across maps
+        this._callbackCounter = 0;
+        this._callbacks = new Map();
+        // internal id -> geojson layer map
+        // TODO: Make geojson functionality an extension on the normal geojson layer
```

Comment:
I think that we could extend L.geoJSON instead of putting this functionality in the wrapper, but it works well enough this way for now.